lib/kubeclient/common.rb

```diff
@@ -425,6 +425,7 @@ def api
     def load_entities
       @entities = {}
       fetch_entities['resources'].each do |resource|
+        resource = Kubeclient::Common::Compatibility31.compatible_resource(resource)
```

Comment:
@zakiva it's late here so don't assume that this works (or makes sense :blush:)... but I would have expected something as:

```ruby
fetch_entities['resources'].each do |resource|
  next if resource['name'].include?('/')
  resource['kind'] = Compatibility31.resource_kind(resource['name']) if resource['kind'].nil?
...
```
Instead of `nil?` you can also think to use `empty?` (I always forget the nits... anyway it would be nice to catch also when it's an empty string).

@moolitayer please check if the above makes sense.
For example, the check `next if resource['name'].include?('/')` is fully valid and working on 3.1 as well?